instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/engine/RuleParserTest.java

```diff
@@ -149,38 +175,62 @@ void testConf3() {
     assertThat(metricDef).isNotNull();
-    assertThat(metricDef.getMetricExtractors()).hasSize(3);
 
     assertThat(metricDef.getMetricExtractors())
+        .hasSize(3)
         .anySatisfy(
             m -> {
               assertThat(m.getMetricValueExtractor().getAttributeName()).isEqualTo("A.b");
-              assertThat(m.getAttributes()).hasSize(3);
-
-              MetricInfo mb1 = m.getInfo();
-              assertThat(mb1.getMetricName()).isEqualTo("PREFIX.METRIC_NAME1");
-              assertThat(mb1.getDescription()).isEqualTo("DESCRIPTION1");
-              assertThat(mb1.getUnit()).isEqualTo("UNIT1");
-              assertThat(mb1.getType()).isEqualTo(MetricInfo.Type.COUNTER);
+              assertThat(m.getAttributes())
+                  .hasSize(3)
+                  .extracting("attributeName")
+                  .contains("LABEL_KEY1", "LABEL_KEY2", "LABEL_KEY3");
+
+              MetricInfo metricInfo = m.getInfo();
+              assertThat(metricInfo.getMetricName()).isEqualTo("PREFIX.METRIC_NAME1");
+              assertThat(metricInfo.getDescription()).isEqualTo("DESCRIPTION1");
+              assertThat(metricInfo.getUnit()).isEqualTo("UNIT1");
+              assertThat(metricInfo.getType()).isEqualTo(MetricInfo.Type.COUNTER);
+            })
+        .anySatisfy(
+            m -> {
+              assertThat(m.getMetricValueExtractor().getAttributeName()).isEqualTo("ATTRIBUTE2");
+              assertThat(m.getAttributes())
+                  .hasSize(2)
+                  .extracting("attributeName")
+                  .contains("LABEL_KEY1", "LABEL_KEY2");
+
+              MetricInfo metricInfo = m.getInfo();
+              assertThat(metricInfo.getMetricName()).isEqualTo("PREFIX.METRIC_NAME2");
+              assertThat(metricInfo.getDescription()).isEqualTo("DESCRIPTION2");
+              assertThat(metricInfo.getUnit()).isEqualTo("UNIT2");
             })
         .anySatisfy(
             m -> {
               assertThat(m.getMetricValueExtractor().getAttributeName()).isEqualTo("ATTRIBUTE3");
 
-              MetricInfo mb3 = m.getInfo();
-              assertThat(mb3.getMetricName()).isEqualTo("PREFIX.ATTRIBUTE3");
+              MetricInfo metricInfo = m.getInfo();
+              assertThat(metricInfo.getMetricName()).isEqualTo("PREFIX.ATTRIBUTE3");
+              assertThat(metricInfo.getDescription()).isNull();
+
               // syntax extension - defining a default unit and type
-              assertThat(mb3.getType()).isEqualTo(MetricInfo.Type.UPDOWNCOUNTER);
-              assertThat(mb3.getUnit()).isEqualTo("DEFAULT_UNIT");
+              assertThat(metricInfo.getType())
+                  .describedAs("default type should match jmx rule definition")
+                  .isEqualTo(jmxDef.getMetricType());
+              assertThat(metricInfo.getUnit())
+                  .describedAs("default unit should match jmx rule definition")
+                  .isEqualTo(jmxDef.getUnit());
```

Comment:
[for reviewer] this is mostly adding extra test assertions, not related to adding state metrics.